docs/reference/mapping/runtime.asciidoc

```diff
@@ -747,6 +747,243 @@ address.
+reported data, you define the following fields in your mapping for
+`my-index`:
+
+[source,console]
+----
+PUT my-index/
+{
+  "mappings": {
+    "properties": {
+      "timestamp": {
+        "type": "date"
+      },
+      "temperature": {
+        "type": "long"
+      },
+      "voltage": {
+        "type": "double"
+      },
+      "node": {
+        "type": "keyword"
+      }
+    }
+  }
+}
+----
+
+You then bulk index some sample data from your sensors. This data includes
+`voltage` readings for each sensor:
+
+[source,console]
+----
+POST my-index/_bulk?refresh=true
+{"index":{}}
+{"timestamp": 1516729294000, "temperature": 200, "voltage": 5.2, "node": "a"}
+{"index":{}}
+{"timestamp": 1516642894000, "temperature": 201, "voltage": 5.8, "node": "b"}
+{"index":{}}
+{"timestamp": 1516556494000, "temperature": 202, "voltage": 5.1, "node": "a"}
+{"index":{}}
+{"timestamp": 1516470094000, "temperature": 198, "voltage": 5.6, "node": "b"}
+{"index":{}}
+{"timestamp": 1516383694000, "temperature": 200, "voltage": 4.2, "node": "c"}
+{"index":{}}
+{"timestamp": 1516297294000, "temperature": 202, "voltage": 4.0, "node": "c"}
+----
+// TEST[continued]
+
+After talking to a few site engineers, you realize that the sensors should
+be reporting at least _double_ the current values, but potentially higher.
+You create a runtime field named `voltage_reading` that retrieves the current
+voltage and multiplies it by `2`:
+
+[source,console]
+----
+PUT my-index/_mapping
+{
+  "runtime": {
+    "voltage_reading": {
+      "type": "long",
+      "script": {
+        "source": """
+          for (double v : doc['voltage'])
+          emit((long)(v * params['multiplier']))
+          """,
+        "params": {
+          "multiplier": 2
+        }
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+
+You retrieve the calculated values using the <<search-fields,`fields`>>
+parameter on the `_search` API:
+
+[source,console]
+----
+GET my-index/_search
+{
+  "fields": [
+    "voltage_reading",
+    "node"
+    ]
+}
+----
+// TEST[continued]
+
+After reviewing the sensor data and running some tests, you determine that the
+multiplier for reported sensor data should be `4`. To gain greater performance,
+you decide to index the `voltage_reading` runtime field with the new
+`multiplier` parameter.
+
+In a new index named `my-index-00001`, copy the `voltage_reading` runtime field
+definition into the mappings of the new index. It's that simple!
+
+[source,console]
+----
+PUT my-index-00001/
+{
+  "mappings": {
+    "properties": {
+      "timestamp": {
+        "type": "date"
+      },
+      "temperature": {
+        "type": "long"
+      },
+      "voltage": {
+        "type": "double"
+      },
+      "node": {
+        "type": "keyword"
+      },
+      "voltage_reading": {
+        "type": "long",
+        "script": {
+          "source": """
+            for (double v : doc['voltage'])
+            emit((long)(v * params['multiplier']))
```

Comment:
Same comment about cast and type of runtime field.

Also, I wonder if we should just do `emit(doc['voltage'].value * params['multiplier'])`. In this story sensors only make a single `voltage` reading. Its nice to describe how you do more than one but it makes the scripts longer. I don't really know how to balance those considerations. Certainly a few months ago I'd have said "do the loop everywhere", but now, :shrug: .

Reply:
Shorter, more succinct scripts are easier to understand, and I think we want to show users how to write good scripts. I'm in favor of shortening to:

```
emit(doc['voltage'].value * params['multiplier'])
```